### PR TITLE
smtp/mime: Restrict file name lengths

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #4585

This commit places restrictions on the length of the file name specified
in attachments (`name=` or `filename=`) to `NAME_MAX`. Names exceeding
these limits will be truncated and processing will continue with the
truncated name.

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [3409](https://redmine.openinfosecfoundation.org/issues/3265)

Describe changes:
- Simplified length checking

### CODE REVIEW

File: src/util-decode-mime.h [hunk 1]
```diff
@@ -52,14 +52,15 @@
...
-#define ANOM_INVALID_QP          2  /* invalid qouted-printable chars */
+#define ANOM_INVALID_QP          2  /* invalid quoted-printable chars */
 #define ANOM_LONG_HEADER_NAME    4  /* header is abnormally long */
 #define ANOM_LONG_HEADER_VALUE   8  /* header value is abnormally long
                                      * (includes multi-line) */
 #define ANOM_LONG_LINE          16  /* Lines that exceed 998 octets */
 #define ANOM_LONG_ENC_LINE      32  /* Lines that exceed 76 octets */
 #define ANOM_MALFORMED_MSG      64  /* Misc msg format errors found */
 #define ANOM_LONG_BOUNDARY     128  /* Boundary too long */
+#define ANOM_LONG_FILENAME     256  /* filename truncated*/
```
Comment (catenacyber) [inline src/util-decode-mime.h]: nit missing space

File: src/util-decode-mime.c [hunk 2]
```diff
@@ -1912,6 +1947,11 @@ static int ProcessMimeHeaders(const uint8_t *buf, uint32_t len,
                 }
                 memcpy(entity->filename, bptr, blen);
                 entity->filename_len = blen;
+
+                if (truncated_name) {
+                    state->stack->top->data->anomaly_flags |= ANOM_LONG_FILENAME;
```
Comment (catenacyber) [inline src/util-decode-mime.c]: Why do we need both `anomaly_flags` ?
It may be worth a comment

Comment (jlucovsky) [reply src/util-decode-mime.c]: That's the existing pattern; I'm not sure why `state->stack->top`'s flags are modified.

**Comment by jlucovsky** [comment]
Continued in #4615

**Pull request closed without merging**